{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "688c978a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Installazione pacchetto in ‘/usr/local/lib/R/site-library’\n",
      "(perché ‘lib’ non è specificato)\n",
      "\n",
      "R[write to console]: apertura URL 'https://cloud.r-project.org/src/contrib/BiocManager_1.30.18.tar.gz'\n",
      "\n",
      "R[write to console]: Content type 'application/x-gzip'\n",
      "R[write to console]:  length 289602 bytes (282 KB)\n",
      "\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: =\n",
      "R[write to console]: \n",
      "\n",
      "R[write to console]: downloaded 282 KB\n",
      "\n",
      "\n",
      "* installing *source* package ‘BiocManager’ ...\n",
      "** pacchetto ‘BiocManager’ aperto con successo con controllo somme MD5\n",
      "** using staged installation\n",
      "** R\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Avvertimento in file(con, \"r\"):\n",
      "  URL 'https://bioconductor.org/config.yaml': lo status era 'Couldn't resolve host name'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (BiocManager)\n",
      "R[write to console]: \n",
      "\n",
      "R[write to console]: \n",
      "R[write to console]: I pacchetti scaricati con il codice sorgente sono in\n",
      "\t‘/tmp/RtmpSfLF9F/downloaded_packages’\n",
      "R[write to console]: \n",
      "R[write to console]: \n",
      "\n",
      "R[write to console]: 'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cloud.r-project.org\n",
      "\n",
      "\n",
      "R[write to console]: Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.1 (2022-06-23)\n",
      "\n",
      "R[write to console]: Installation paths not writeable, unable to update packages\n",
      "  path: /usr/lib/R/library\n",
      "  packages:\n",
      "    foreign, MASS, nnet\n",
      "\n",
      "R[write to console]: 'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cloud.r-project.org\n",
      "\n",
      "\n",
      "R[write to console]: Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.1 (2022-06-23)\n",
      "\n",
      "R[write to console]: Installation paths not writeable, unable to update packages\n",
      "  path: /usr/lib/R/library\n",
      "  packages:\n",
      "    foreign, MASS, nnet\n",
      "\n",
      "R[write to console]: 'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cloud.r-project.org\n",
      "\n",
      "\n",
      "R[write to console]: Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.1 (2022-06-23)\n",
      "\n",
      "R[write to console]: Installation paths not writeable, unable to update packages\n",
      "  path: /usr/lib/R/library\n",
      "  packages:\n",
      "    foreign, MASS, nnet\n",
      "\n",
      "R[write to console]: 'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'?repositories' for details\n",
      "\n",
      "replacement repositories:\n",
      "    CRAN: https://cloud.r-project.org\n",
      "\n",
      "\n",
      "R[write to console]: Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.1 (2022-06-23)\n",
      "\n",
      "R[write to console]: Installation paths not writeable, unable to update packages\n",
      "  path: /usr/lib/R/library\n",
      "  packages:\n",
      "    foreign, MASS, nnet\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: MultiAssayExperiment\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: SummarizedExperiment\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: MatrixGenerics\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: matrixStats\n",
      "\n",
      "R[write to console]: \n",
      "Caricamento pacchetto: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "R[write to console]: I seguenti oggetti sono mascherati da ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: GenomicRanges\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: stats4\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: BiocGenerics\n",
      "\n",
      "R[write to console]: \n",
      "Caricamento pacchetto: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "R[write to console]: I seguenti oggetti sono mascherati da ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "R[write to console]: I seguenti oggetti sono mascherati da ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: S4Vectors\n",
      "\n",
      "R[write to console]: \n",
      "Caricamento pacchetto: ‘S4Vectors’\n",
      "\n",
      "\n",
      "R[write to console]: I seguenti oggetti sono mascherati da ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: IRanges\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: GenomeInfoDb\n",
      "\n",
      "R[write to console]: Caricamento del pacchetto richiesto: Biobase\n",
      "\n",
      "R[write to console]: Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "R[write to console]: \n",
      "Caricamento pacchetto: ‘Biobase’\n",
      "\n",
      "\n",
      "R[write to console]: Il seguente oggetto è mascherato da ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "R[write to console]: I seguenti oggetti sono mascherati da ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "R[write to console]: \n",
      "Caricamento pacchetto: ‘TCGAutils’\n",
      "\n",
      "\n",
      "R[write to console]: Il seguente oggetto è mascherato da ‘package:MultiAssayExperiment’:\n",
      "\n",
      "    splitAssays\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <span>StrVector with 21 elements.</span>\n",
       "        <table>\n",
       "        <tbody>\n",
       "          <tr>\n",
       "          \n",
       "            <td>\n",
       "            'TCGAutils'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'curatedT...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'MultiAss...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            ...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'datasets'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'methods'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'base'\n",
       "            </td>\n",
       "          \n",
       "          </tr>\n",
       "        </tbody>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<rpy2.robjects.vectors.StrVector object at 0x7fafa580c2c0> [RTYPES.STRSXP]\n",
       "R classes: ('character',)\n",
       "['TCGAutils', 'curatedT..., 'MultiAss..., 'Summariz..., ..., 'utils', 'datasets', 'methods', 'base']"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from rpy2.robjects.packages import importr\n",
    "import rpy2\n",
    "from rpy2 import robjects\n",
    "from rpy2.robjects import pandas2ri\n",
    "import rpy2.robjects as ro\n",
    "\n",
    "utils = importr('utils')\n",
    "\n",
    "utils.install_packages('BiocManager')\n",
    "robjects.r('''\n",
    "    BiocManager::install(version = '3.15')\n",
    "    BiocManager::install(\"curatedTCGAData\")\n",
    "    BiocManager::install('RaggedExperiment')\n",
    "    BiocManager::install('TCGAutils')\n",
    "    library(curatedTCGAData)\n",
    "    library(MultiAssayExperiment)\n",
    "    library(TCGAutils)\n",
    "    ''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e218ed98",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File: /tmp/RtmpSfLF9F/Rtxt4e26b2d3b3f98\n",
      "curatedTCGAData        package:curatedTCGAData         R Documentation\n",
      "\n",
      "\n",
      "\n",
      "_\bC_\br_\be_\ba_\bt_\be _\ba _\bM_\bu_\bl_\bt_\bi_\bA_\bs_\bs_\ba_\by_\bE_\bx_\bp_\be_\br_\bi_\bm_\be_\bn_\bt _\bf_\br_\bo_\bm _\bs_\bp_\be_\bc_\bi_\bf_\bi_\bc _\ba_\bs_\bs_\ba_\by_\bs _\ba_\bn_\bd _\bc_\bo_\bh_\bo_\br_\bt_\bs\n",
      "\n",
      "\n",
      "\n",
      "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
      "\n",
      "\n",
      "\n",
      "     curatedTCGAData assembles data on-the-fly from ExperimentHub to\n",
      "\n",
      "     provide cohesive MultiAssayExperiment container objects. All the\n",
      "\n",
      "     user has to do is to provide TCGA disease code(s) and assay types.\n",
      "\n",
      "     It is highly recommended to use the companion package ‘TCGAutils’,\n",
      "\n",
      "     developed to work with TCGA data specifically from\n",
      "\n",
      "     ‘curatedTCGAData’ and some flat files.\n",
      "\n",
      "\n",
      "\n",
      "_\bU_\bs_\ba_\bg_\be:\n",
      "\n",
      "\n",
      "\n",
      "     curatedTCGAData(\n",
      "\n",
      "       diseaseCode = \"*\",\n",
      "\n",
      "       assays = \"*\",\n",
      "\n",
      "       version,\n",
      "\n",
      "       dry.run = TRUE,\n",
      "\n",
      "       verbose = TRUE,\n",
      "\n",
      "       ...\n",
      "\n",
      "     )\n",
      "\n",
      "     \n",
      "\n",
      "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
      "\n",
      "\n",
      "\n",
      "diseaseCode: character() A vector of TCGA cancer cohort codes (e.g.,\n",
      "\n",
      "          ‘COAD’)\n",
      "\n",
      "\n",
      "\n",
      "  assays: character() A vector of TCGA assays, glob matches allowed;\n",
      "\n",
      "          see below for more details\n",
      "\n",
      "\n",
      "\n",
      " version: character(1) Either 1.1.38 or 2.0.1 indicating the data\n",
      "\n",
      "          version to obtain from ‘ExperimentHub’. Version 2.0.1\n",
      "\n",
      "          includes various improvements as well as the addition of the\n",
      "\n",
      "          ‘RNASeq2Gene’ assay. See ‘version’ section details.\n",
      "\n",
      "\n",
      "\n",
      " dry.run: logical(1) Whether to return the dataset names before actual\n",
      "\n",
      "          download (default TRUE)\n",
      "\n",
      "\n",
      "\n",
      " verbose: logical(1) Whether to show the dataset currenlty being\n",
      "\n",
      "          (down)loaded (default TRUE)\n",
      "\n",
      "\n",
      "\n",
      "     ...: Additional arguments passed on to the ‘ExperimentHub’\n",
      "\n",
      "          constructor\n",
      "\n",
      "\n",
      "\n",
      "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
      "\n",
      "\n",
      "\n",
      "     This function will check against available resources in\n",
      "\n",
      "     ExperimentHub. Only the latest runDate (\"2016-01-28\") is\n",
      "\n",
      "     supported. Use the ‘dry.run = FALSE’ to download remote datasets\n",
      "\n",
      "     and build an integrative MultiAssayExperiment object. For a list\n",
      "\n",
      "     of 'diseaseCodes', see the curatedTCGAData-package help page.\n",
      "\n",
      "\n",
      "\n",
      "_\bV_\ba_\bl_\bu_\be:\n",
      "\n",
      "\n",
      "\n",
      "     a MultiAssayExperiment of the specified assays and cancer codes or\n",
      "\n",
      "     informative data.frame of resources when ‘dry.run’ is ‘TRUE’\n",
      "\n",
      "\n",
      "\n",
      "_\bA_\bv_\ba_\bi_\bl_\ba_\bb_\bl_\be _\bA_\bs_\bs_\ba_\by_\bs:\n",
      "\n",
      "\n",
      "\n",
      "     Below is a list of partial ExperimentList assay names and their\n",
      "\n",
      "     respective description. These assays can be entered as part of the\n",
      "\n",
      "     ‘assays’ argument in the main function. Partial glob matches are\n",
      "\n",
      "     allowed such as: ‘'CN*'’ for \"CNASeq\", \"CNASNP\", \"CNVSNP\" assays.\n",
      "\n",
      "     Credit: Ludwig G.\n",
      "\n",
      "\n",
      "\n",
      "     ExperimentList data types   Description\n",
      "\n",
      "     ----------------------------------------------------------------------------\n",
      "\n",
      "     SummarizedExperiment*\n",
      "\n",
      "       RNASeqGene                Gene expression values\n",
      "\n",
      "       RNASeq2Gene               RSEM TPM gene expression values\n",
      "\n",
      "       RNASeq2GeneNorm           Upper quartile normalized RSEM TPM gene\n",
      "\n",
      "                                 expression values\n",
      "\n",
      "       miRNAArray                Probe-level  miRNA expression values\n",
      "\n",
      "       miRNASeqGene              Gene-level log2 RPM miRNA expression values\n",
      "\n",
      "       mRNAArray                 Unified gene-level mRNA expression values\n",
      "\n",
      "       mRNAArray_huex            Gene-level mRNA expression values from Affymetrix\n",
      "\n",
      "                                 Human Exon Array\n",
      "\n",
      "       mRNAArray_TX_g4502a       Gene-level mRNA expression values from Agilent\n",
      "\n",
      "                                 244K Array\n",
      "\n",
      "       mRNAArray_TX_ht_hg_u133a  Gene-level mRNA expression values from Affymetrix\n",
      "\n",
      "                                 Human Genome U133 Array\n",
      "\n",
      "       GISTIC_AllByGene          Gene-level GISTIC2 copy number values\n",
      "\n",
      "       GISTIC_ThresholdedByGene  Gene-level GISTIC2 thresholded discrete copy\n",
      "\n",
      "                                 number values\n",
      "\n",
      "       RPPAArray                 Reverse Phase Protein Array normalized protein\n",
      "\n",
      "                                 expression values\n",
      "\n",
      "     RangedSummarizedExperiment\n",
      "\n",
      "       GISTIC_Peaks              GISTIC2 thresholded discrete copy number values\n",
      "\n",
      "                                 in recurrent peak regions\n",
      "\n",
      "     SummarizedExperiment with HDF5Array DelayedMatrix\n",
      "\n",
      "       Methylation_methyl27      Probe-level methylation beta values from Illumina\n",
      "\n",
      "                                 HumanMethylation 27K BeadChip\n",
      "\n",
      "       Methylation_methyl450     Probe-level methylation beta values from Infinium\n",
      "\n",
      "                                 HumanMethylation 450K BeadChip\n",
      "\n",
      "     RaggedExperiment\n",
      "\n",
      "       CNASNP                    Segmented somatic Copy Number Alteration calls\n",
      "\n",
      "                                 from SNP array\n",
      "\n",
      "       CNVSNP                    Segmented germline Copy Number Variant calls from\n",
      "\n",
      "                                 SNP Array\n",
      "\n",
      "       CNASeq                    Segmented somatic Copy Number Alteration calls\n",
      "\n",
      "                                 from low pass DNA Sequencing\n",
      "\n",
      "       Mutation*                 Somatic mutations calls\n",
      "\n",
      "       CNACGH_CGH_hg_244a        Segmented somatic Copy Number Alteration calls\n",
      "\n",
      "                                 from CGH Agilent Microarray 244A\n",
      "\n",
      "       CNACGH_CGH_hg_415k_g4124a Segmented somatic Copy Number Alteration calls\n",
      "\n",
      "                                 from CGH Agilent Microarray 415K\n",
      "\n",
      "     * All can be converted to RangedSummarizedExperiment (except RPPAArray) with\n",
      "\n",
      "     TCGAutils\n",
      "\n",
      "     \n",
      "\n",
      "_\bv_\be_\br_\bs_\bi_\bo_\bn:\n",
      "\n",
      "\n",
      "\n",
      "     The new version 2.0.1 includes various improvements including an\n",
      "\n",
      "     additional assay that provides ‘RNASeq2Gene’ data as RSEM TPM gene\n",
      "\n",
      "     expression values (issue #38). Additional changes include genomic\n",
      "\n",
      "     information for ‘RaggedExperiment’ type data objects where '37' is\n",
      "\n",
      "     now 'GRCh37' as reported in issue #40. Datasets (e.g., OV, GBM)\n",
      "\n",
      "     that contain multiple assays that could be merged are now provided\n",
      "\n",
      "     as merged assays (issue #27). We corrected an issue where\n",
      "\n",
      "     ‘mRNAArray’ assays were returning ‘DataFrame’s instead of ‘matrix’\n",
      "\n",
      "     type data (issue #31). Version 1.1.38 provides the original run of\n",
      "\n",
      "     ‘curatedTCGAData’ and is provided due to legacy reasons.\n",
      "\n",
      "\n",
      "\n",
      "_\bS_\be_\be _\bA_\bl_\bs_\bo:\n",
      "\n",
      "\n",
      "\n",
      "     curatedTCGAData-package\n",
      "\n",
      "\n",
      "\n",
      "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
      "\n",
      "\n",
      "\n",
      "     curatedTCGAData(\n",
      "\n",
      "         diseaseCode = c(\"GBM\", \"ACC\"), assays = \"CNASNP\", version = \"2.0.1\"\n",
      "\n",
      "     )\n",
      "\n",
      "     \n",
      "\n",
      "     curatedTCGAData(\"BRCA\", \"GISTIC*\", \"2.0.1\")\n",
      "\n",
      "     \n",
      "\n",
      "---\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(robjects.r('''?curatedTCGAData'''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "41ab5070",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: snapshotDate(): 2022-04-26\n",
      "\n",
      "R[write to console]: See '?curatedTCGAData' for 'diseaseCode' and 'assays' inputs\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     ah_id                                       title file_size\n",
      "1   EH4737                         ACC_CNASNP-20160128    0.8 Mb\n",
      "2   EH4738                         ACC_CNVSNP-20160128    0.2 Mb\n",
      "3   EH4740               ACC_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "4   EH4741                   ACC_GISTIC_Peaks-20160128      0 Mb\n",
      "5   EH4742       ACC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "6   EH4744             ACC_Methylation-20160128_assays  239.2 Mb\n",
      "7   EH4745                 ACC_Methylation-20160128_se      6 Mb\n",
      "8   EH4746                   ACC_miRNASeqGene-20160128    0.1 Mb\n",
      "9   EH4747                       ACC_Mutation-20160128    0.7 Mb\n",
      "10  EH4748                    ACC_RNASeq2Gene-20160128    2.7 Mb\n",
      "11  EH4749                ACC_RNASeq2GeneNorm-20160128      4 Mb\n",
      "12  EH4750                      ACC_RPPAArray-20160128    0.1 Mb\n",
      "13  EH4752                        BLCA_CNASeq-20160128    0.3 Mb\n",
      "14  EH4753                        BLCA_CNASNP-20160128    4.3 Mb\n",
      "15  EH4754                        BLCA_CNVSNP-20160128    1.1 Mb\n",
      "16  EH4756              BLCA_GISTIC_AllByGene-20160128    0.5 Mb\n",
      "17  EH4757                  BLCA_GISTIC_Peaks-20160128      0 Mb\n",
      "18  EH4758      BLCA_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "19  EH4760            BLCA_Methylation-20160128_assays 1300.4 Mb\n",
      "20  EH4761                BLCA_Methylation-20160128_se      6 Mb\n",
      "21  EH4762                  BLCA_miRNASeqGene-20160128    0.3 Mb\n",
      "22  EH4763                      BLCA_Mutation-20160128    3.7 Mb\n",
      "23  EH4764                   BLCA_RNASeq2Gene-20160128   14.7 Mb\n",
      "24  EH4765               BLCA_RNASeq2GeneNorm-20160128   21.9 Mb\n",
      "25  EH4766                    BLCA_RNASeqGene-20160128    2.4 Mb\n",
      "26  EH4767                     BLCA_RPPAArray-20160128    0.5 Mb\n",
      "27  EH4769                        BRCA_CNASeq-20160128      0 Mb\n",
      "28  EH4770                        BRCA_CNASNP-20160128    9.8 Mb\n",
      "29  EH4771                        BRCA_CNVSNP-20160128    2.8 Mb\n",
      "30  EH4773              BRCA_GISTIC_AllByGene-20160128    1.2 Mb\n",
      "31  EH4774                  BRCA_GISTIC_Peaks-20160128      0 Mb\n",
      "32  EH4775      BRCA_GISTIC_ThresholdedByGene-20160128    0.3 Mb\n",
      "33  EH4777   BRCA_Methylation_methyl27-20160128_assays   60.7 Mb\n",
      "34  EH4778       BRCA_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "35  EH4779  BRCA_Methylation_methyl450-20160128_assays 2646.4 Mb\n",
      "36  EH4780      BRCA_Methylation_methyl450-20160128_se      6 Mb\n",
      "37  EH4781                  BRCA_miRNASeqGene-20160128    0.6 Mb\n",
      "38  EH4782                     BRCA_mRNAArray-20160128   27.3 Mb\n",
      "39  EH4783                      BRCA_Mutation-20160128    4.5 Mb\n",
      "40  EH4784                   BRCA_RNASeq2Gene-20160128   43.1 Mb\n",
      "41  EH4785               BRCA_RNASeq2GeneNorm-20160128   64.5 Mb\n",
      "42  EH4786                    BRCA_RNASeqGene-20160128     30 Mb\n",
      "43  EH4787                     BRCA_RPPAArray-20160128    1.5 Mb\n",
      "44  EH4789                        CESC_CNASeq-20160128    0.1 Mb\n",
      "45  EH4790                        CESC_CNASNP-20160128    2.3 Mb\n",
      "46  EH4791                        CESC_CNVSNP-20160128    0.6 Mb\n",
      "47  EH4793              CESC_GISTIC_AllByGene-20160128    0.4 Mb\n",
      "48  EH4794                  CESC_GISTIC_Peaks-20160128      0 Mb\n",
      "49  EH4795      CESC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "50  EH4797            CESC_Methylation-20160128_assays  932.9 Mb\n",
      "51  EH4798                CESC_Methylation-20160128_se      6 Mb\n",
      "52  EH4799                  CESC_miRNASeqGene-20160128    0.2 Mb\n",
      "53  EH4800                      CESC_Mutation-20160128      2 Mb\n",
      "54  EH4801                   CESC_RNASeq2Gene-20160128   10.8 Mb\n",
      "55  EH4802               CESC_RNASeq2GeneNorm-20160128   16.2 Mb\n",
      "56  EH4803                     CESC_RPPAArray-20160128    0.3 Mb\n",
      "57  EH4805                        CHOL_CNASNP-20160128    0.4 Mb\n",
      "58  EH4806                        CHOL_CNVSNP-20160128    0.1 Mb\n",
      "59  EH4808              CHOL_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "60  EH4809                  CHOL_GISTIC_Peaks-20160128      0 Mb\n",
      "61  EH4810      CHOL_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "62  EH4812            CHOL_Methylation-20160128_assays  134.1 Mb\n",
      "63  EH4813                CHOL_Methylation-20160128_se      6 Mb\n",
      "64  EH4814                  CHOL_miRNASeqGene-20160128      0 Mb\n",
      "65  EH4815                      CHOL_Mutation-20160128    0.2 Mb\n",
      "66  EH4816                   CHOL_RNASeq2Gene-20160128    1.6 Mb\n",
      "67  EH4817               CHOL_RNASeq2GeneNorm-20160128    2.3 Mb\n",
      "68  EH4818                     CHOL_RPPAArray-20160128      0 Mb\n",
      "69  EH4820                        COAD_CNASeq-20160128    0.3 Mb\n",
      "70  EH4821                        COAD_CNASNP-20160128    3.9 Mb\n",
      "71  EH4822                        COAD_CNVSNP-20160128    0.9 Mb\n",
      "72  EH4824              COAD_GISTIC_AllByGene-20160128    0.4 Mb\n",
      "73  EH4825                  COAD_GISTIC_Peaks-20160128      0 Mb\n",
      "74  EH4826      COAD_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "75  EH4828   COAD_Methylation_methyl27-20160128_assays   35.8 Mb\n",
      "76  EH4829       COAD_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "77  EH4830  COAD_Methylation_methyl450-20160128_assays    996 Mb\n",
      "78  EH4831      COAD_Methylation_methyl450-20160128_se      6 Mb\n",
      "79  EH4832                  COAD_miRNASeqGene-20160128    0.2 Mb\n",
      "80  EH4833                     COAD_mRNAArray-20160128      8 Mb\n",
      "81  EH4834                      COAD_Mutation-20160128    1.2 Mb\n",
      "82  EH4835                   COAD_RNASeq2Gene-20160128   11.3 Mb\n",
      "83  EH4836    COAD_RNASeq2GeneNorm_illuminaga-20160128    8.8 Mb\n",
      "84  EH4837 COAD_RNASeq2GeneNorm_illuminahiseq-20160128   16.8 Mb\n",
      "85  EH4838                    COAD_RNASeqGene-20160128    0.4 Mb\n",
      "86  EH4839                     COAD_RPPAArray-20160128    0.5 Mb\n",
      "87  EH4841                        DLBC_CNASNP-20160128    0.4 Mb\n",
      "88  EH4842                        DLBC_CNVSNP-20160128    0.1 Mb\n",
      "89  EH4844              DLBC_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "90  EH4845                  DLBC_GISTIC_Peaks-20160128      0 Mb\n",
      "91  EH4846      DLBC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "92  EH4848            DLBC_Methylation-20160128_assays  143.6 Mb\n",
      "93  EH4849                DLBC_Methylation-20160128_se      6 Mb\n",
      "94  EH4850                  DLBC_miRNASeqGene-20160128      0 Mb\n",
      "95  EH4851                      DLBC_Mutation-20160128    0.7 Mb\n",
      "96  EH4852                   DLBC_RNASeq2Gene-20160128    1.7 Mb\n",
      "97  EH4853               DLBC_RNASeq2GeneNorm-20160128    2.5 Mb\n",
      "98  EH4854                     DLBC_RPPAArray-20160128      0 Mb\n",
      "99  EH4856                        ESCA_CNASeq-20160128    0.2 Mb\n",
      "100 EH4857                        ESCA_CNASNP-20160128    1.8 Mb\n",
      "101 EH4858                        ESCA_CNVSNP-20160128    0.7 Mb\n",
      "102 EH4860              ESCA_GISTIC_AllByGene-20160128    0.4 Mb\n",
      "103 EH4861                  ESCA_GISTIC_Peaks-20160128      0 Mb\n",
      "104 EH4862      ESCA_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "105 EH4864            ESCA_Methylation-20160128_assays  604.7 Mb\n",
      "106 EH4865                ESCA_Methylation-20160128_se      6 Mb\n",
      "107 EH4866                  ESCA_miRNASeqGene-20160128    0.2 Mb\n",
      "108 EH4867                      ESCA_Mutation-20160128    2.8 Mb\n",
      "109 EH4868                   ESCA_RNASeq2Gene-20160128    7.3 Mb\n",
      "110 EH4869               ESCA_RNASeq2GeneNorm-20160128   10.8 Mb\n",
      "111 EH4870                    ESCA_RNASeqGene-20160128      8 Mb\n",
      "112 EH4871                     ESCA_RPPAArray-20160128    0.2 Mb\n",
      "113 EH4873             GBM_CNACGH_CGH_hg_244a-20160128    0.7 Mb\n",
      "114 EH4874      GBM_CNACGH_CGH_hg_415k_g4124a-20160128    0.5 Mb\n",
      "115 EH4875                         GBM_CNASNP-20160128    5.2 Mb\n",
      "116 EH4876                         GBM_CNVSNP-20160128    1.5 Mb\n",
      "117 EH4878               GBM_GISTIC_AllByGene-20160128    0.6 Mb\n",
      "118 EH4879                   GBM_GISTIC_Peaks-20160128      0 Mb\n",
      "119 EH4880       GBM_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "120 EH4882    GBM_Methylation_methyl27-20160128_assays   50.4 Mb\n",
      "121 EH4883        GBM_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "122 EH4884   GBM_Methylation_methyl450-20160128_assays  460.9 Mb\n",
      "123 EH4885       GBM_Methylation_methyl450-20160128_se      6 Mb\n",
      "124 EH4886                     GBM_miRNAArray-20160128    2.1 Mb\n",
      "125 EH4887                   GBM_miRNASeqGene-20160128      0 Mb\n",
      "126 EH4888                 GBM_mRNAArray_huex-20160128   56.2 Mb\n",
      "127 EH4889            GBM_mRNAArray_TX_g4502a-20160128     28 Mb\n",
      "128 EH4890       GBM_mRNAArray_TX_ht_hg_u133a-20160128   44.7 Mb\n",
      "129 EH4891                       GBM_Mutation-20160128    2.1 Mb\n",
      "130 EH4892                    GBM_RNASeq2Gene-20160128    5.8 Mb\n",
      "131 EH4893                GBM_RNASeq2GeneNorm-20160128    8.5 Mb\n",
      "132 EH4894                      GBM_RPPAArray-20160128    0.4 Mb\n",
      "133 EH4896                        HNSC_CNASeq-20160128    0.3 Mb\n",
      "134 EH4897                        HNSC_CNASNP-20160128    4.2 Mb\n",
      "135 EH4898                        HNSC_CNVSNP-20160128    1.1 Mb\n",
      "136 EH4900              HNSC_GISTIC_AllByGene-20160128    0.5 Mb\n",
      "137 EH4901                  HNSC_GISTIC_Peaks-20160128      0 Mb\n",
      "138 EH4902      HNSC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "139 EH4904            HNSC_Methylation-20160128_assays 1736.2 Mb\n",
      "140 EH4905                HNSC_Methylation-20160128_se      6 Mb\n",
      "141 EH4906                  HNSC_miRNASeqGene-20160128    0.4 Mb\n",
      "142 EH4907                      HNSC_Mutation-20160128    4.8 Mb\n",
      "143 EH4908                   HNSC_RNASeq2Gene-20160128   19.8 Mb\n",
      "144 EH4909               HNSC_RNASeq2GeneNorm-20160128   29.5 Mb\n",
      "145 EH4910                    HNSC_RNASeqGene-20160128   10.3 Mb\n",
      "146 EH4911                     HNSC_RPPAArray-20160128    0.3 Mb\n",
      "147 EH4913                        KICH_CNASNP-20160128    0.5 Mb\n",
      "148 EH4914                        KICH_CNVSNP-20160128    0.1 Mb\n",
      "149 EH4916              KICH_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "150 EH4917                  KICH_GISTIC_Peaks-20160128      0 Mb\n",
      "151 EH4918      KICH_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "152 EH4920            KICH_Methylation-20160128_assays  197.4 Mb\n",
      "153 EH4921                KICH_Methylation-20160128_se      6 Mb\n",
      "154 EH4922                  KICH_miRNASeqGene-20160128    0.1 Mb\n",
      "155 EH4923                      KICH_Mutation-20160128    0.1 Mb\n",
      "156 EH4924                   KICH_RNASeq2Gene-20160128    3.3 Mb\n",
      "157 EH4925               KICH_RNASeq2GeneNorm-20160128    4.9 Mb\n",
      "158 EH4926                     KICH_RPPAArray-20160128    0.1 Mb\n",
      "159 EH4928                        KIRC_CNASNP-20160128    4.1 Mb\n",
      "160 EH4929                        KIRC_CNVSNP-20160128    0.8 Mb\n",
      "161 EH4931              KIRC_GISTIC_AllByGene-20160128    0.4 Mb\n",
      "162 EH4932                  KIRC_GISTIC_Peaks-20160128      0 Mb\n",
      "163 EH4933      KIRC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "164 EH4935   KIRC_Methylation_methyl27-20160128_assays   73.9 Mb\n",
      "165 EH4936       KIRC_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "166 EH4937  KIRC_Methylation_methyl450-20160128_assays 1436.5 Mb\n",
      "167 EH4938      KIRC_Methylation_methyl450-20160128_se      6 Mb\n",
      "168 EH4939                  KIRC_miRNASeqGene-20160128    0.2 Mb\n",
      "169 EH4940                     KIRC_mRNAArray-20160128    3.4 Mb\n",
      "170 EH4941                      KIRC_Mutation-20160128    0.4 Mb\n",
      "171 EH4942                   KIRC_RNASeq2Gene-20160128   21.6 Mb\n",
      "172 EH4943               KIRC_RNASeq2GeneNorm-20160128   32.1 Mb\n",
      "173 EH4944                    KIRC_RNASeqGene-20160128   18.1 Mb\n",
      "174 EH4945                     KIRC_RPPAArray-20160128    0.8 Mb\n",
      "175 EH4947                        KIRP_CNASNP-20160128    2.6 Mb\n",
      "176 EH4948                        KIRP_CNVSNP-20160128    0.5 Mb\n",
      "177 EH4950              KIRP_GISTIC_AllByGene-20160128    0.3 Mb\n",
      "178 EH4951                  KIRP_GISTIC_Peaks-20160128      0 Mb\n",
      "179 EH4952      KIRP_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "180 EH4954   KIRP_Methylation_methyl27-20160128_assays    3.7 Mb\n",
      "181 EH4955       KIRP_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "182 EH4956  KIRP_Methylation_methyl450-20160128_assays  958.9 Mb\n",
      "183 EH4957      KIRP_Methylation_methyl450-20160128_se      6 Mb\n",
      "184 EH4958                  KIRP_miRNASeqGene-20160128    0.2 Mb\n",
      "185 EH4959                     KIRP_mRNAArray-20160128    0.9 Mb\n",
      "186 EH4960                      KIRP_Mutation-20160128    0.5 Mb\n",
      "187 EH4961                   KIRP_RNASeq2Gene-20160128   11.3 Mb\n",
      "188 EH4962               KIRP_RNASeq2GeneNorm-20160128   16.7 Mb\n",
      "189 EH4963                    KIRP_RNASeqGene-20160128    0.6 Mb\n",
      "190 EH4964                     KIRP_RPPAArray-20160128    0.3 Mb\n",
      "191 EH4966                        LAML_CNASNP-20160128    7.6 Mb\n",
      "192 EH4967                        LAML_CNVSNP-20160128    0.3 Mb\n",
      "193 EH4969              LAML_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "194 EH4970                  LAML_GISTIC_Peaks-20160128      0 Mb\n",
      "195 EH4971      LAML_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "196 EH4973   LAML_Methylation_methyl27-20160128_assays   34.2 Mb\n",
      "197 EH4974       LAML_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "198 EH4975  LAML_Methylation_methyl450-20160128_assays  578.6 Mb\n",
      "199 EH4976      LAML_Methylation_methyl450-20160128_se      6 Mb\n",
      "200 EH4977                      LAML_Mutation-20160128    0.2 Mb\n",
      "201 EH4978                   LAML_RNASeq2Gene-20160128    5.4 Mb\n",
      "202 EH4979               LAML_RNASeq2GeneNorm-20160128      8 Mb\n",
      "203 EH4980                    LAML_RNASeqGene-20160128     NA Mb\n",
      "204 EH4982                         LGG_CNASeq-20160128    0.1 Mb\n",
      "205 EH4983                         LGG_CNASNP-20160128    3.3 Mb\n",
      "206 EH4984                         LGG_CNVSNP-20160128    0.8 Mb\n",
      "207 EH4986               LGG_GISTIC_AllByGene-20160128    0.4 Mb\n",
      "208 EH4987                   LGG_GISTIC_Peaks-20160128      0 Mb\n",
      "209 EH4988       LGG_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "210 EH4990             LGG_Methylation-20160128_assays 1579.4 Mb\n",
      "211 EH4991                 LGG_Methylation-20160128_se      6 Mb\n",
      "212 EH4992                   LGG_miRNASeqGene-20160128    0.4 Mb\n",
      "213 EH4993                      LGG_mRNAArray-20160128    1.6 Mb\n",
      "214 EH4994                       LGG_Mutation-20160128    0.2 Mb\n",
      "215 EH4995                    LGG_RNASeq2Gene-20160128     19 Mb\n",
      "216 EH4996                LGG_RNASeq2GeneNorm-20160128   28.3 Mb\n",
      "217 EH4997                      LGG_RPPAArray-20160128    0.6 Mb\n",
      "218 EH4999                        LIHC_CNASNP-20160128    3.1 Mb\n",
      "219 EH5000                        LIHC_CNVSNP-20160128      1 Mb\n",
      "220 EH5002              LIHC_GISTIC_AllByGene-20160128    0.5 Mb\n",
      "221 EH5003                  LIHC_GISTIC_Peaks-20160128      0 Mb\n",
      "222 EH5004      LIHC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "223 EH5006            LIHC_Methylation-20160128_assays 1284.7 Mb\n",
      "224 EH5007                LIHC_Methylation-20160128_se      6 Mb\n",
      "225 EH5008                  LIHC_miRNASeqGene-20160128    0.3 Mb\n",
      "226 EH5009                      LIHC_Mutation-20160128    0.9 Mb\n",
      "227 EH5010                   LIHC_RNASeq2Gene-20160128   13.8 Mb\n",
      "228 EH5011               LIHC_RNASeq2GeneNorm-20160128   20.6 Mb\n",
      "229 EH5012                    LIHC_RNASeqGene-20160128    0.9 Mb\n",
      "230 EH5013                     LIHC_RPPAArray-20160128    0.3 Mb\n",
      "231 EH5015                        LUAD_CNASeq-20160128    3.8 Mb\n",
      "232 EH5016                        LUAD_CNASNP-20160128    4.2 Mb\n",
      "233 EH5017                        LUAD_CNVSNP-20160128    1.2 Mb\n",
      "234 EH5019              LUAD_GISTIC_AllByGene-20160128    0.6 Mb\n",
      "235 EH5020                  LUAD_GISTIC_Peaks-20160128      0 Mb\n",
      "236 EH5021      LUAD_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "237 EH5023   LUAD_Methylation_methyl27-20160128_assays   15.8 Mb\n",
      "238 EH5024       LUAD_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "239 EH5025  LUAD_Methylation_methyl450-20160128_assays 1471.1 Mb\n",
      "240 EH5026      LUAD_Methylation_methyl450-20160128_se      6 Mb\n",
      "241 EH5027                  LUAD_miRNASeqGene-20160128    0.4 Mb\n",
      "242 EH5028                     LUAD_mRNAArray-20160128    1.6 Mb\n",
      "243 EH5029                      LUAD_Mutation-20160128    6.7 Mb\n",
      "244 EH5030                   LUAD_RNASeq2Gene-20160128   20.1 Mb\n",
      "245 EH5031               LUAD_RNASeq2GeneNorm-20160128   29.8 Mb\n",
      "246 EH5032                    LUAD_RNASeqGene-20160128    5.6 Mb\n",
      "247 EH5033                     LUAD_RPPAArray-20160128    0.6 Mb\n",
      "248 EH5035                        LUSC_CNACGH-20160128    0.7 Mb\n",
      "249 EH5036                        LUSC_CNASNP-20160128    4.7 Mb\n",
      "250 EH5037                        LUSC_CNVSNP-20160128    1.4 Mb\n",
      "251 EH5039              LUSC_GISTIC_AllByGene-20160128    0.6 Mb\n",
      "252 EH5040                  LUSC_GISTIC_Peaks-20160128      0 Mb\n",
      "253 EH5041      LUSC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "254 EH5043   LUSC_Methylation_methyl27-20160128_assays   28.3 Mb\n",
      "255 EH5044       LUSC_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "256 EH5045  LUSC_Methylation_methyl450-20160128_assays 1233.9 Mb\n",
      "257 EH5046      LUSC_Methylation_methyl450-20160128_se      6 Mb\n",
      "258 EH5047                  LUSC_miRNASeqGene-20160128    0.3 Mb\n",
      "259 EH5048                LUSC_mRNAArray_huex-20160128   14.7 Mb\n",
      "260 EH5049           LUSC_mRNAArray_TX_g4502a-20160128    7.2 Mb\n",
      "261 EH5050      LUSC_mRNAArray_TX_ht_hg_u133a-20160128   11.3 Mb\n",
      "262 EH5051                      LUSC_Mutation-20160128    5.7 Mb\n",
      "263 EH5052                   LUSC_RNASeq2Gene-20160128   19.7 Mb\n",
      "264 EH5053               LUSC_RNASeq2GeneNorm-20160128   29.2 Mb\n",
      "265 EH5054                    LUSC_RNASeqGene-20160128    8.4 Mb\n",
      "266 EH5055                     LUSC_RPPAArray-20160128    0.5 Mb\n",
      "267 EH5057                        MESO_CNASNP-20160128    0.8 Mb\n",
      "268 EH5058                        MESO_CNVSNP-20160128    0.2 Mb\n",
      "269 EH5060              MESO_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "270 EH5061                  MESO_GISTIC_Peaks-20160128      0 Mb\n",
      "271 EH5062      MESO_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "272 EH5064            MESO_Methylation-20160128_assays  260.3 Mb\n",
      "273 EH5065                MESO_Methylation-20160128_se      6 Mb\n",
      "274 EH5066                  MESO_miRNASeqGene-20160128    0.1 Mb\n",
      "275 EH5067                   MESO_RNASeq2Gene-20160128    3.1 Mb\n",
      "276 EH5068               MESO_RNASeq2GeneNorm-20160128    4.6 Mb\n",
      "277 EH5069                     MESO_RPPAArray-20160128    0.1 Mb\n",
      "278 EH5071              OV_CNACGH_CGH_hg_244a-20160128    1.1 Mb\n",
      "279 EH5072       OV_CNACGH_CGH_hg_415k_g4124a-20160128    2.2 Mb\n",
      "280 EH5073                          OV_CNASNP-20160128    8.2 Mb\n",
      "281 EH5074                          OV_CNVSNP-20160128    2.7 Mb\n",
      "282 EH5076                OV_GISTIC_AllByGene-20160128    1.1 Mb\n",
      "283 EH5077                    OV_GISTIC_Peaks-20160128      0 Mb\n",
      "284 EH5078        OV_GISTIC_ThresholdedByGene-20160128    0.3 Mb\n",
      "285 EH5080     OV_Methylation_methyl27-20160128_assays  104.5 Mb\n",
      "286 EH5081         OV_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "287 EH5082    OV_Methylation_methyl450-20160128_assays     30 Mb\n",
      "288 EH5083        OV_Methylation_methyl450-20160128_se      6 Mb\n",
      "289 EH5084                      OV_miRNAArray-20160128    3.2 Mb\n",
      "290 EH5085                    OV_miRNASeqGene-20160128    0.3 Mb\n",
      "291 EH5086                  OV_mRNAArray_huex-20160128   75.1 Mb\n",
      "292 EH5087             OV_mRNAArray_TX_g4502a-20160128   26.8 Mb\n",
      "293 EH5088        OV_mRNAArray_TX_ht_hg_u133a-20160128   44.5 Mb\n",
      "294 EH5089                        OV_Mutation-20160128    0.5 Mb\n",
      "295 EH5090                     OV_RNASeq2Gene-20160128   11.2 Mb\n",
      "296 EH5091                 OV_RNASeq2GeneNorm-20160128   16.7 Mb\n",
      "297 EH5092                      OV_RNASeqGene-20160128   10.3 Mb\n",
      "298 EH5093                       OV_RPPAArray-20160128    0.7 Mb\n",
      "299 EH5095                        PAAD_CNASNP-20160128    1.7 Mb\n",
      "300 EH5096                        PAAD_CNVSNP-20160128    0.4 Mb\n",
      "301 EH5098              PAAD_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "302 EH5099                  PAAD_GISTIC_Peaks-20160128      0 Mb\n",
      "303 EH5100      PAAD_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "304 EH5102            PAAD_Methylation-20160128_assays  583.2 Mb\n",
      "305 EH5103                PAAD_Methylation-20160128_se      6 Mb\n",
      "306 EH5104                  PAAD_miRNASeqGene-20160128    0.1 Mb\n",
      "307 EH5105                      PAAD_Mutation-20160128    6.8 Mb\n",
      "308 EH5106                   PAAD_RNASeq2Gene-20160128    6.5 Mb\n",
      "309 EH5107               PAAD_RNASeq2GeneNorm-20160128    9.6 Mb\n",
      "310 EH5108                     PAAD_RPPAArray-20160128    0.2 Mb\n",
      "311 EH5110                        PCPG_CNASNP-20160128    2.6 Mb\n",
      "312 EH5111                        PCPG_CNVSNP-20160128    0.3 Mb\n",
      "313 EH5113              PCPG_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "314 EH5114                  PCPG_GISTIC_Peaks-20160128      0 Mb\n",
      "315 EH5115      PCPG_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "316 EH5117            PCPG_Methylation-20160128_assays  560.2 Mb\n",
      "317 EH5118                PCPG_Methylation-20160128_se      6 Mb\n",
      "318 EH5119                  PCPG_miRNASeqGene-20160128    0.2 Mb\n",
      "319 EH5120                      PCPG_Mutation-20160128    0.6 Mb\n",
      "320 EH5121                   PCPG_RNASeq2Gene-20160128    6.5 Mb\n",
      "321 EH5122               PCPG_RNASeq2GeneNorm-20160128    9.6 Mb\n",
      "322 EH5123                     PCPG_RPPAArray-20160128    0.1 Mb\n",
      "323 EH5125                        PRAD_CNASeq-20160128    0.2 Mb\n",
      "324 EH5126                        PRAD_CNASNP-20160128    4.9 Mb\n",
      "325 EH5127                        PRAD_CNVSNP-20160128    1.2 Mb\n",
      "326 EH5129              PRAD_GISTIC_AllByGene-20160128    0.5 Mb\n",
      "327 EH5130                  PRAD_GISTIC_Peaks-20160128      0 Mb\n",
      "328 EH5131      PRAD_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "329 EH5133            PRAD_Methylation-20160128_assays 1641.3 Mb\n",
      "330 EH5134                PRAD_Methylation-20160128_se      6 Mb\n",
      "331 EH5135                  PRAD_miRNASeqGene-20160128    0.4 Mb\n",
      "332 EH5136                      PRAD_Mutation-20160128    1.4 Mb\n",
      "333 EH5137                   PRAD_RNASeq2Gene-20160128   19.5 Mb\n",
      "334 EH5138               PRAD_RNASeq2GeneNorm-20160128   28.9 Mb\n",
      "335 EH5139                     PRAD_RPPAArray-20160128    0.5 Mb\n",
      "336 EH5141                        READ_CNASeq-20160128    0.4 Mb\n",
      "337 EH5142                        READ_CNASNP-20160128    1.4 Mb\n",
      "338 EH5143                        READ_CNVSNP-20160128    0.4 Mb\n",
      "339 EH5145              READ_GISTIC_AllByGene-20160128    0.3 Mb\n",
      "340 EH5146                  READ_GISTIC_Peaks-20160128      0 Mb\n",
      "341 EH5147      READ_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "342 EH5149   READ_Methylation_methyl27-20160128_assays     13 Mb\n",
      "343 EH5150       READ_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "344 EH5151  READ_Methylation_methyl450-20160128_assays  317.5 Mb\n",
      "345 EH5152      READ_Methylation_methyl450-20160128_se      6 Mb\n",
      "346 EH5153                  READ_miRNASeqGene-20160128    0.1 Mb\n",
      "347 EH5154                     READ_mRNAArray-20160128    3.4 Mb\n",
      "348 EH5155                      READ_Mutation-20160128    0.4 Mb\n",
      "349 EH5156                   READ_RNASeq2Gene-20160128    3.7 Mb\n",
      "350 EH5157    READ_RNASeq2GeneNorm_illuminaga-20160128    3.5 Mb\n",
      "351 EH5158 READ_RNASeq2GeneNorm_illuminahiseq-20160128    5.5 Mb\n",
      "352 EH5159                    READ_RNASeqGene-20160128    2.1 Mb\n",
      "353 EH5160                     READ_RPPAArray-20160128    0.2 Mb\n",
      "354 EH5162                        SARC_CNASNP-20160128    3.1 Mb\n",
      "355 EH5163                        SARC_CNVSNP-20160128    1.1 Mb\n",
      "356 EH5165              SARC_GISTIC_AllByGene-20160128    0.5 Mb\n",
      "357 EH5166                  SARC_GISTIC_Peaks-20160128      0 Mb\n",
      "358 EH5167      SARC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "359 EH5169            SARC_Methylation-20160128_assays    805 Mb\n",
      "360 EH5170                SARC_Methylation-20160128_se      6 Mb\n",
      "361 EH5171                  SARC_miRNASeqGene-20160128    0.2 Mb\n",
      "362 EH5172                      SARC_Mutation-20160128    1.1 Mb\n",
      "363 EH5173                   SARC_RNASeq2Gene-20160128    9.2 Mb\n",
      "364 EH5174               SARC_RNASeq2GeneNorm-20160128   13.6 Mb\n",
      "365 EH5175                     SARC_RPPAArray-20160128    0.3 Mb\n",
      "366 EH5177                        SKCM_CNASeq-20160128    0.3 Mb\n",
      "367 EH5178                        SKCM_CNASNP-20160128    3.8 Mb\n",
      "368 EH5179                        SKCM_CNVSNP-20160128    1.1 Mb\n",
      "369 EH5181              SKCM_GISTIC_AllByGene-20160128    0.4 Mb\n",
      "370 EH5182                  SKCM_GISTIC_Peaks-20160128      0 Mb\n",
      "371 EH5183      SKCM_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "372 EH5185            SKCM_Methylation-20160128_assays 1421.4 Mb\n",
      "373 EH5186                SKCM_Methylation-20160128_se      6 Mb\n",
      "374 EH5187                  SKCM_miRNASeqGene-20160128    0.4 Mb\n",
      "375 EH5188                      SKCM_Mutation-20160128   26.7 Mb\n",
      "376 EH5189                   SKCM_RNASeq2Gene-20160128   16.4 Mb\n",
      "377 EH5190               SKCM_RNASeq2GeneNorm-20160128   24.5 Mb\n",
      "378 EH5191                     SKCM_RPPAArray-20160128    0.5 Mb\n",
      "379 EH5193                        STAD_CNASeq-20160128    0.3 Mb\n",
      "380 EH5194                        STAD_CNASNP-20160128    3.8 Mb\n",
      "381 EH5195                        STAD_CNVSNP-20160128    1.2 Mb\n",
      "382 EH5197              STAD_GISTIC_AllByGene-20160128    0.6 Mb\n",
      "383 EH5198                  STAD_GISTIC_Peaks-20160128      0 Mb\n",
      "384 EH5199      STAD_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "385 EH5201   STAD_Methylation_methyl27-20160128_assays     13 Mb\n",
      "386 EH5202       STAD_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "387 EH5203  STAD_Methylation_methyl450-20160128_assays 1187.8 Mb\n",
      "388 EH5204      STAD_Methylation_methyl450-20160128_se      6 Mb\n",
      "389 EH5205                  STAD_miRNASeqGene-20160128    0.3 Mb\n",
      "390 EH5206                      STAD_Mutation-20160128   13.2 Mb\n",
      "391 EH5207                   STAD_RNASeq2Gene-20160128   16.1 Mb\n",
      "392 EH5208               STAD_RNASeq2GeneNorm-20160128   23.9 Mb\n",
      "393 EH5209                    STAD_RNASeqGene-20160128   10.7 Mb\n",
      "394 EH5210                     STAD_RPPAArray-20160128    0.5 Mb\n",
      "395 EH5212                        TGCT_CNASNP-20160128    1.2 Mb\n",
      "396 EH5213                        TGCT_CNVSNP-20160128    0.3 Mb\n",
      "397 EH5215              TGCT_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "398 EH5216                  TGCT_GISTIC_Peaks-20160128      0 Mb\n",
      "399 EH5217      TGCT_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "400 EH5219            TGCT_Methylation-20160128_assays  416.6 Mb\n",
      "401 EH5220                TGCT_Methylation-20160128_se      6 Mb\n",
      "402 EH5221                  TGCT_miRNASeqGene-20160128    0.1 Mb\n",
      "403 EH5222                      TGCT_Mutation-20160128    0.5 Mb\n",
      "404 EH5223                   TGCT_RNASeq2Gene-20160128      5 Mb\n",
      "405 EH5224               TGCT_RNASeq2GeneNorm-20160128    7.4 Mb\n",
      "406 EH5225                     TGCT_RPPAArray-20160128    0.2 Mb\n",
      "407 EH5227                        THCA_CNASeq-20160128      0 Mb\n",
      "408 EH5228                        THCA_CNASNP-20160128    3.1 Mb\n",
      "409 EH5229                        THCA_CNVSNP-20160128    0.5 Mb\n",
      "410 EH5231              THCA_GISTIC_AllByGene-20160128    0.3 Mb\n",
      "411 EH5232                  THCA_GISTIC_Peaks-20160128      0 Mb\n",
      "412 EH5233      THCA_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "413 EH5235            THCA_Methylation-20160128_assays 1693.8 Mb\n",
      "414 EH5236                THCA_Methylation-20160128_se      6 Mb\n",
      "415 EH5237                  THCA_miRNASeqGene-20160128    0.4 Mb\n",
      "416 EH5238                      THCA_Mutation-20160128    0.9 Mb\n",
      "417 EH5239                   THCA_RNASeq2Gene-20160128   20.1 Mb\n",
      "418 EH5240               THCA_RNASeq2GeneNorm-20160128   30.1 Mb\n",
      "419 EH5241                    THCA_RNASeqGene-20160128    0.2 Mb\n",
      "420 EH5242                     THCA_RPPAArray-20160128    0.3 Mb\n",
      "421 EH5244                        THYM_CNASNP-20160128    0.9 Mb\n",
      "422 EH5245                        THYM_CNVSNP-20160128    0.2 Mb\n",
      "423 EH5247              THYM_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "424 EH5248                  THYM_GISTIC_Peaks-20160128      0 Mb\n",
      "425 EH5249      THYM_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "426 EH5251            THYM_Methylation-20160128_assays  376.4 Mb\n",
      "427 EH5252                THYM_Methylation-20160128_se      6 Mb\n",
      "428 EH5253                  THYM_miRNASeqGene-20160128    0.1 Mb\n",
      "429 EH5254                      THYM_Mutation-20160128    0.2 Mb\n",
      "430 EH5255                   THYM_RNASeq2Gene-20160128    4.4 Mb\n",
      "431 EH5256               THYM_RNASeq2GeneNorm-20160128    6.4 Mb\n",
      "432 EH5257                     THYM_RPPAArray-20160128    0.1 Mb\n",
      "433 EH5259                        UCEC_CNASeq-20160128    0.3 Mb\n",
      "434 EH5260                        UCEC_CNASNP-20160128    5.4 Mb\n",
      "435 EH5261                        UCEC_CNVSNP-20160128    1.3 Mb\n",
      "436 EH5263              UCEC_GISTIC_AllByGene-20160128    0.6 Mb\n",
      "437 EH5264                  UCEC_GISTIC_Peaks-20160128      0 Mb\n",
      "438 EH5265      UCEC_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "439 EH5267   UCEC_Methylation_methyl27-20160128_assays   20.9 Mb\n",
      "440 EH5268       UCEC_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "441 EH5269  UCEC_Methylation_methyl450-20160128_assays 1394.3 Mb\n",
      "442 EH5270      UCEC_Methylation_methyl450-20160128_se      6 Mb\n",
      "443 EH5271                  UCEC_miRNASeqGene-20160128    0.4 Mb\n",
      "444 EH5272                     UCEC_mRNAArray-20160128    2.6 Mb\n",
      "445 EH5273                      UCEC_Mutation-20160128    4.6 Mb\n",
      "446 EH5274                   UCEC_RNASeq2Gene-20160128    6.6 Mb\n",
      "447 EH5275    UCEC_RNASeq2GeneNorm_illuminaga-20160128   18.4 Mb\n",
      "448 EH5276 UCEC_RNASeq2GeneNorm_illuminahiseq-20160128    9.8 Mb\n",
      "449 EH5277                    UCEC_RNASeqGene-20160128    7.7 Mb\n",
      "450 EH5278                     UCEC_RPPAArray-20160128    0.7 Mb\n",
      "451 EH5280                         UCS_CNASNP-20160128    0.5 Mb\n",
      "452 EH5281                         UCS_CNVSNP-20160128    0.2 Mb\n",
      "453 EH5283               UCS_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "454 EH5284                   UCS_GISTIC_Peaks-20160128      0 Mb\n",
      "455 EH5285       UCS_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "456 EH5287             UCS_Methylation-20160128_assays  170.7 Mb\n",
      "457 EH5288                 UCS_Methylation-20160128_se      6 Mb\n",
      "458 EH5289                   UCS_miRNASeqGene-20160128    0.1 Mb\n",
      "459 EH5290                       UCS_Mutation-20160128    1.3 Mb\n",
      "460 EH5291                    UCS_RNASeq2Gene-20160128    2.1 Mb\n",
      "461 EH5292                UCS_RNASeq2GeneNorm-20160128    3.1 Mb\n",
      "462 EH5293                      UCS_RPPAArray-20160128    0.1 Mb\n",
      "463 EH5295                         UVM_CNASeq-20160128    0.1 Mb\n",
      "464 EH5296                         UVM_CNASNP-20160128    0.6 Mb\n",
      "465 EH5297                         UVM_CNVSNP-20160128    0.1 Mb\n",
      "466 EH5299               UVM_GISTIC_AllByGene-20160128    0.2 Mb\n",
      "467 EH5300                   UVM_GISTIC_Peaks-20160128      0 Mb\n",
      "468 EH5301       UVM_GISTIC_ThresholdedByGene-20160128    0.2 Mb\n",
      "469 EH5303             UVM_Methylation-20160128_assays  238.9 Mb\n",
      "470 EH5304                 UVM_Methylation-20160128_se      6 Mb\n",
      "471 EH5305                   UVM_miRNASeqGene-20160128    0.1 Mb\n",
      "472 EH5306                       UVM_Mutation-20160128    0.9 Mb\n",
      "473 EH5307                    UVM_RNASeq2Gene-20160128    2.7 Mb\n",
      "474 EH5308                UVM_RNASeq2GeneNorm-20160128      4 Mb\n",
      "475 EH5309                      UVM_RPPAArray-20160128      0 Mb\n",
      "                    rdataclass rdatadateadded rdatadateremoved\n",
      "1             RaggedExperiment     2021-01-27             <NA>\n",
      "2             RaggedExperiment     2021-01-27             <NA>\n",
      "3         SummarizedExperiment     2021-01-27             <NA>\n",
      "4   RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "5         SummarizedExperiment     2021-01-27             <NA>\n",
      "6         SummarizedExperiment     2021-01-27             <NA>\n",
      "7             RaggedExperiment     2021-01-27             <NA>\n",
      "8         SummarizedExperiment     2021-01-27             <NA>\n",
      "9         SummarizedExperiment     2021-01-27             <NA>\n",
      "10        SummarizedExperiment     2021-01-27             <NA>\n",
      "11                      DFrame     2021-01-27             <NA>\n",
      "12        SummarizedExperiment     2021-01-27             <NA>\n",
      "13            RaggedExperiment     2021-01-27             <NA>\n",
      "14            RaggedExperiment     2021-01-27             <NA>\n",
      "15            RaggedExperiment     2021-01-27             <NA>\n",
      "16        SummarizedExperiment     2021-01-27             <NA>\n",
      "17  RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "18        SummarizedExperiment     2021-01-27             <NA>\n",
      "19        SummarizedExperiment     2021-01-27             <NA>\n",
      "20            RaggedExperiment     2021-01-27             <NA>\n",
      "21        SummarizedExperiment     2021-01-27             <NA>\n",
      "22        SummarizedExperiment     2021-01-27             <NA>\n",
      "23        SummarizedExperiment     2021-01-27             <NA>\n",
      "24        SummarizedExperiment     2021-01-27             <NA>\n",
      "25                      DFrame     2021-01-27             <NA>\n",
      "26        SummarizedExperiment     2021-01-27             <NA>\n",
      "27            RaggedExperiment     2021-01-27             <NA>\n",
      "28            RaggedExperiment     2021-01-27             <NA>\n",
      "29            RaggedExperiment     2021-01-27             <NA>\n",
      "30        SummarizedExperiment     2021-01-27             <NA>\n",
      "31  RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "32        SummarizedExperiment     2021-01-27             <NA>\n",
      "33        SummarizedExperiment     2021-01-27             <NA>\n",
      "34        SummarizedExperiment     2021-01-27             <NA>\n",
      "35            RaggedExperiment     2021-01-27             <NA>\n",
      "36        SummarizedExperiment     2021-01-27             <NA>\n",
      "37        SummarizedExperiment     2021-01-27             <NA>\n",
      "38        SummarizedExperiment     2021-01-27             <NA>\n",
      "39        SummarizedExperiment     2021-01-27             <NA>\n",
      "40                      DFrame     2021-01-27             <NA>\n",
      "41        SummarizedExperiment     2021-01-27             <NA>\n",
      "42        SummarizedExperiment     2021-01-27             <NA>\n",
      "43        SummarizedExperiment     2021-01-27             <NA>\n",
      "44            RaggedExperiment     2021-01-27             <NA>\n",
      "45            RaggedExperiment     2021-01-27             <NA>\n",
      "46            RaggedExperiment     2021-01-27             <NA>\n",
      "47        SummarizedExperiment     2021-01-27             <NA>\n",
      "48  RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "49        SummarizedExperiment     2021-01-27             <NA>\n",
      "50        SummarizedExperiment     2021-01-27             <NA>\n",
      "51            RaggedExperiment     2021-01-27             <NA>\n",
      "52        SummarizedExperiment     2021-01-27             <NA>\n",
      "53        SummarizedExperiment     2021-01-27             <NA>\n",
      "54        SummarizedExperiment     2021-01-27             <NA>\n",
      "55                      DFrame     2021-01-27             <NA>\n",
      "56        SummarizedExperiment     2021-01-27             <NA>\n",
      "57            RaggedExperiment     2021-01-27             <NA>\n",
      "58            RaggedExperiment     2021-01-27             <NA>\n",
      "59        SummarizedExperiment     2021-01-27             <NA>\n",
      "60  RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "61        SummarizedExperiment     2021-01-27             <NA>\n",
      "62        SummarizedExperiment     2021-01-27             <NA>\n",
      "63            RaggedExperiment     2021-01-27             <NA>\n",
      "64        SummarizedExperiment     2021-01-27             <NA>\n",
      "65        SummarizedExperiment     2021-01-27             <NA>\n",
      "66        SummarizedExperiment     2021-01-27             <NA>\n",
      "67                      DFrame     2021-01-27             <NA>\n",
      "68        SummarizedExperiment     2021-01-27             <NA>\n",
      "69            RaggedExperiment     2021-01-27             <NA>\n",
      "70            RaggedExperiment     2021-01-27             <NA>\n",
      "71            RaggedExperiment     2021-01-27             <NA>\n",
      "72        SummarizedExperiment     2021-01-27             <NA>\n",
      "73  RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "74        SummarizedExperiment     2021-01-27             <NA>\n",
      "75        SummarizedExperiment     2021-01-27             <NA>\n",
      "76        SummarizedExperiment     2021-01-27             <NA>\n",
      "77            RaggedExperiment     2021-01-27             <NA>\n",
      "78        SummarizedExperiment     2021-01-27             <NA>\n",
      "79        SummarizedExperiment     2021-01-27             <NA>\n",
      "80        SummarizedExperiment     2021-01-27             <NA>\n",
      "81        SummarizedExperiment     2021-01-27             <NA>\n",
      "82        SummarizedExperiment     2021-01-27             <NA>\n",
      "83                      DFrame     2021-01-27             <NA>\n",
      "84        SummarizedExperiment     2021-01-27             <NA>\n",
      "85        SummarizedExperiment     2021-01-27             <NA>\n",
      "86        SummarizedExperiment     2021-01-27             <NA>\n",
      "87            RaggedExperiment     2021-01-27             <NA>\n",
      "88            RaggedExperiment     2021-01-27             <NA>\n",
      "89        SummarizedExperiment     2021-01-27             <NA>\n",
      "90  RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "91        SummarizedExperiment     2021-01-27             <NA>\n",
      "92        SummarizedExperiment     2021-01-27             <NA>\n",
      "93            RaggedExperiment     2021-01-27             <NA>\n",
      "94        SummarizedExperiment     2021-01-27             <NA>\n",
      "95        SummarizedExperiment     2021-01-27             <NA>\n",
      "96        SummarizedExperiment     2021-01-27             <NA>\n",
      "97                      DFrame     2021-01-27             <NA>\n",
      "98        SummarizedExperiment     2021-01-27             <NA>\n",
      "99            RaggedExperiment     2021-01-27             <NA>\n",
      "100           RaggedExperiment     2021-01-27             <NA>\n",
      "101           RaggedExperiment     2021-01-27             <NA>\n",
      "102       SummarizedExperiment     2021-01-27             <NA>\n",
      "103 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "104       SummarizedExperiment     2021-01-27             <NA>\n",
      "105       SummarizedExperiment     2021-01-27             <NA>\n",
      "106           RaggedExperiment     2021-01-27             <NA>\n",
      "107       SummarizedExperiment     2021-01-27             <NA>\n",
      "108       SummarizedExperiment     2021-01-27             <NA>\n",
      "109       SummarizedExperiment     2021-01-27             <NA>\n",
      "110       SummarizedExperiment     2021-01-27             <NA>\n",
      "111                     DFrame     2021-01-27             <NA>\n",
      "112       SummarizedExperiment     2021-01-27             <NA>\n",
      "113           RaggedExperiment     2021-01-27             <NA>\n",
      "114           RaggedExperiment     2021-01-27             <NA>\n",
      "115           RaggedExperiment     2021-01-27             <NA>\n",
      "116           RaggedExperiment     2021-01-27             <NA>\n",
      "117       SummarizedExperiment     2021-01-27             <NA>\n",
      "118 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "119       SummarizedExperiment     2021-01-27             <NA>\n",
      "120       SummarizedExperiment     2021-01-27             <NA>\n",
      "121       SummarizedExperiment     2021-01-27             <NA>\n",
      "122       SummarizedExperiment     2021-01-27             <NA>\n",
      "123       SummarizedExperiment     2021-01-27             <NA>\n",
      "124       SummarizedExperiment     2021-01-27             <NA>\n",
      "125           RaggedExperiment     2021-01-27             <NA>\n",
      "126       SummarizedExperiment     2021-01-27             <NA>\n",
      "127       SummarizedExperiment     2021-01-27             <NA>\n",
      "128       SummarizedExperiment     2021-01-27             <NA>\n",
      "129                     DFrame     2021-01-27             <NA>\n",
      "130       SummarizedExperiment     2021-01-27             <NA>\n",
      "131       SummarizedExperiment     2021-01-27             <NA>\n",
      "132       SummarizedExperiment     2021-01-27             <NA>\n",
      "133           RaggedExperiment     2021-01-27             <NA>\n",
      "134           RaggedExperiment     2021-01-27             <NA>\n",
      "135           RaggedExperiment     2021-01-27             <NA>\n",
      "136       SummarizedExperiment     2021-01-27             <NA>\n",
      "137 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "138       SummarizedExperiment     2021-01-27             <NA>\n",
      "139       SummarizedExperiment     2021-01-27             <NA>\n",
      "140           RaggedExperiment     2021-01-27             <NA>\n",
      "141       SummarizedExperiment     2021-01-27             <NA>\n",
      "142       SummarizedExperiment     2021-01-27             <NA>\n",
      "143       SummarizedExperiment     2021-01-27             <NA>\n",
      "144       SummarizedExperiment     2021-01-27             <NA>\n",
      "145                     DFrame     2021-01-27             <NA>\n",
      "146       SummarizedExperiment     2021-01-27             <NA>\n",
      "147           RaggedExperiment     2021-01-27             <NA>\n",
      "148           RaggedExperiment     2021-01-27             <NA>\n",
      "149       SummarizedExperiment     2021-01-27             <NA>\n",
      "150 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "151       SummarizedExperiment     2021-01-27             <NA>\n",
      "152       SummarizedExperiment     2021-01-27             <NA>\n",
      "153           RaggedExperiment     2021-01-27             <NA>\n",
      "154       SummarizedExperiment     2021-01-27             <NA>\n",
      "155       SummarizedExperiment     2021-01-27             <NA>\n",
      "156       SummarizedExperiment     2021-01-27             <NA>\n",
      "157                     DFrame     2021-01-27             <NA>\n",
      "158       SummarizedExperiment     2021-01-27             <NA>\n",
      "159           RaggedExperiment     2021-01-27             <NA>\n",
      "160           RaggedExperiment     2021-01-27             <NA>\n",
      "161       SummarizedExperiment     2021-01-27             <NA>\n",
      "162 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "163       SummarizedExperiment     2021-01-27             <NA>\n",
      "164       SummarizedExperiment     2021-01-27             <NA>\n",
      "165       SummarizedExperiment     2021-01-27             <NA>\n",
      "166           RaggedExperiment     2021-01-27             <NA>\n",
      "167       SummarizedExperiment     2021-01-27             <NA>\n",
      "168       SummarizedExperiment     2021-01-27             <NA>\n",
      "169       SummarizedExperiment     2021-01-27             <NA>\n",
      "170       SummarizedExperiment     2021-01-27             <NA>\n",
      "171                     DFrame     2021-01-27             <NA>\n",
      "172       SummarizedExperiment     2021-01-27             <NA>\n",
      "173       SummarizedExperiment     2021-01-27             <NA>\n",
      "174       SummarizedExperiment     2021-01-27             <NA>\n",
      "175           RaggedExperiment     2021-01-27             <NA>\n",
      "176           RaggedExperiment     2021-01-27             <NA>\n",
      "177       SummarizedExperiment     2021-01-27             <NA>\n",
      "178 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "179       SummarizedExperiment     2021-01-27             <NA>\n",
      "180       SummarizedExperiment     2021-01-27             <NA>\n",
      "181       SummarizedExperiment     2021-01-27             <NA>\n",
      "182           RaggedExperiment     2021-01-27             <NA>\n",
      "183       SummarizedExperiment     2021-01-27             <NA>\n",
      "184       SummarizedExperiment     2021-01-27             <NA>\n",
      "185       SummarizedExperiment     2021-01-27             <NA>\n",
      "186       SummarizedExperiment     2021-01-27             <NA>\n",
      "187                     DFrame     2021-01-27             <NA>\n",
      "188       SummarizedExperiment     2021-01-27             <NA>\n",
      "189       SummarizedExperiment     2021-01-27             <NA>\n",
      "190       SummarizedExperiment     2021-01-27             <NA>\n",
      "191           RaggedExperiment     2021-01-27             <NA>\n",
      "192           RaggedExperiment     2021-01-27             <NA>\n",
      "193       SummarizedExperiment     2021-01-27             <NA>\n",
      "194 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "195       SummarizedExperiment     2021-01-27             <NA>\n",
      "196           RaggedExperiment     2021-01-27             <NA>\n",
      "197       SummarizedExperiment     2021-01-27             <NA>\n",
      "198       SummarizedExperiment     2021-01-27             <NA>\n",
      "199       SummarizedExperiment     2021-01-27             <NA>\n",
      "200                     DFrame     2021-01-27             <NA>\n",
      "201       SummarizedExperiment     2021-01-27             <NA>\n",
      "202       SummarizedExperiment     2021-01-27             <NA>\n",
      "203       SummarizedExperiment     2021-01-27             <NA>\n",
      "204           RaggedExperiment     2021-01-27             <NA>\n",
      "205           RaggedExperiment     2021-01-27             <NA>\n",
      "206           RaggedExperiment     2021-01-27             <NA>\n",
      "207       SummarizedExperiment     2021-01-27             <NA>\n",
      "208 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "209       SummarizedExperiment     2021-01-27             <NA>\n",
      "210       SummarizedExperiment     2021-01-27             <NA>\n",
      "211       SummarizedExperiment     2021-01-27             <NA>\n",
      "212           RaggedExperiment     2021-01-27             <NA>\n",
      "213       SummarizedExperiment     2021-01-27             <NA>\n",
      "214       SummarizedExperiment     2021-01-27             <NA>\n",
      "215       SummarizedExperiment     2021-01-27             <NA>\n",
      "216                     DFrame     2021-01-27             <NA>\n",
      "217       SummarizedExperiment     2021-01-27             <NA>\n",
      "218           RaggedExperiment     2021-01-27             <NA>\n",
      "219           RaggedExperiment     2021-01-27             <NA>\n",
      "220       SummarizedExperiment     2021-01-27             <NA>\n",
      "221 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "222       SummarizedExperiment     2021-01-27             <NA>\n",
      "223       SummarizedExperiment     2021-01-27             <NA>\n",
      "224           RaggedExperiment     2021-01-27             <NA>\n",
      "225       SummarizedExperiment     2021-01-27             <NA>\n",
      "226       SummarizedExperiment     2021-01-27             <NA>\n",
      "227       SummarizedExperiment     2021-01-27             <NA>\n",
      "228       SummarizedExperiment     2021-01-27             <NA>\n",
      "229                     DFrame     2021-01-27             <NA>\n",
      "230       SummarizedExperiment     2021-01-27             <NA>\n",
      "231           RaggedExperiment     2021-01-27             <NA>\n",
      "232           RaggedExperiment     2021-01-27             <NA>\n",
      "233           RaggedExperiment     2021-01-27             <NA>\n",
      "234       SummarizedExperiment     2021-01-27             <NA>\n",
      "235 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "236       SummarizedExperiment     2021-01-27             <NA>\n",
      "237       SummarizedExperiment     2021-01-27             <NA>\n",
      "238       SummarizedExperiment     2021-01-27             <NA>\n",
      "239           RaggedExperiment     2021-01-27             <NA>\n",
      "240       SummarizedExperiment     2021-01-27             <NA>\n",
      "241       SummarizedExperiment     2021-01-27             <NA>\n",
      "242       SummarizedExperiment     2021-01-27             <NA>\n",
      "243       SummarizedExperiment     2021-01-27             <NA>\n",
      "244                     DFrame     2021-01-27             <NA>\n",
      "245       SummarizedExperiment     2021-01-27             <NA>\n",
      "246       SummarizedExperiment     2021-01-27             <NA>\n",
      "247       SummarizedExperiment     2021-01-27             <NA>\n",
      "248           RaggedExperiment     2021-01-27             <NA>\n",
      "249           RaggedExperiment     2021-01-27             <NA>\n",
      "250           RaggedExperiment     2021-01-27             <NA>\n",
      "251       SummarizedExperiment     2021-01-27             <NA>\n",
      "252 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "253       SummarizedExperiment     2021-01-27             <NA>\n",
      "254       SummarizedExperiment     2021-01-27             <NA>\n",
      "255       SummarizedExperiment     2021-01-27             <NA>\n",
      "256       SummarizedExperiment     2021-01-27             <NA>\n",
      "257       SummarizedExperiment     2021-01-27             <NA>\n",
      "258           RaggedExperiment     2021-01-27             <NA>\n",
      "259       SummarizedExperiment     2021-01-27             <NA>\n",
      "260       SummarizedExperiment     2021-01-27             <NA>\n",
      "261       SummarizedExperiment     2021-01-27             <NA>\n",
      "262       SummarizedExperiment     2021-01-27             <NA>\n",
      "263                     DFrame     2021-01-27             <NA>\n",
      "264       SummarizedExperiment     2021-01-27             <NA>\n",
      "265       SummarizedExperiment     2021-01-27             <NA>\n",
      "266       SummarizedExperiment     2021-01-27             <NA>\n",
      "267           RaggedExperiment     2021-01-27             <NA>\n",
      "268           RaggedExperiment     2021-01-27             <NA>\n",
      "269       SummarizedExperiment     2021-01-27             <NA>\n",
      "270 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "271       SummarizedExperiment     2021-01-27             <NA>\n",
      "272       SummarizedExperiment     2021-01-27             <NA>\n",
      "273       SummarizedExperiment     2021-01-27             <NA>\n",
      "274       SummarizedExperiment     2021-01-27             <NA>\n",
      "275       SummarizedExperiment     2021-01-27             <NA>\n",
      "276                     DFrame     2021-01-27             <NA>\n",
      "277       SummarizedExperiment     2021-01-27             <NA>\n",
      "278           RaggedExperiment     2021-01-27             <NA>\n",
      "279           RaggedExperiment     2021-01-27             <NA>\n",
      "280           RaggedExperiment     2021-01-27             <NA>\n",
      "281           RaggedExperiment     2021-01-27             <NA>\n",
      "282       SummarizedExperiment     2021-01-27             <NA>\n",
      "283 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "284       SummarizedExperiment     2021-01-27             <NA>\n",
      "285       SummarizedExperiment     2021-01-27             <NA>\n",
      "286       SummarizedExperiment     2021-01-27             <NA>\n",
      "287       SummarizedExperiment     2021-01-27             <NA>\n",
      "288       SummarizedExperiment     2021-01-27             <NA>\n",
      "289       SummarizedExperiment     2021-01-27             <NA>\n",
      "290           RaggedExperiment     2021-01-27             <NA>\n",
      "291       SummarizedExperiment     2021-01-27             <NA>\n",
      "292       SummarizedExperiment     2021-01-27             <NA>\n",
      "293       SummarizedExperiment     2021-01-27             <NA>\n",
      "294       SummarizedExperiment     2021-01-27             <NA>\n",
      "295                     DFrame     2021-01-27             <NA>\n",
      "296       SummarizedExperiment     2021-01-27             <NA>\n",
      "297       SummarizedExperiment     2021-01-27             <NA>\n",
      "298       SummarizedExperiment     2021-01-27             <NA>\n",
      "299           RaggedExperiment     2021-01-27             <NA>\n",
      "300           RaggedExperiment     2021-01-27             <NA>\n",
      "301       SummarizedExperiment     2021-01-27             <NA>\n",
      "302 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "303       SummarizedExperiment     2021-01-27             <NA>\n",
      "304       SummarizedExperiment     2021-01-27             <NA>\n",
      "305           RaggedExperiment     2021-01-27             <NA>\n",
      "306       SummarizedExperiment     2021-01-27             <NA>\n",
      "307       SummarizedExperiment     2021-01-27             <NA>\n",
      "308       SummarizedExperiment     2021-01-27             <NA>\n",
      "309                     DFrame     2021-01-27             <NA>\n",
      "310       SummarizedExperiment     2021-01-27             <NA>\n",
      "311           RaggedExperiment     2021-01-27             <NA>\n",
      "312           RaggedExperiment     2021-01-27             <NA>\n",
      "313       SummarizedExperiment     2021-01-27             <NA>\n",
      "314 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "315       SummarizedExperiment     2021-01-27             <NA>\n",
      "316       SummarizedExperiment     2021-01-27             <NA>\n",
      "317           RaggedExperiment     2021-01-27             <NA>\n",
      "318       SummarizedExperiment     2021-01-27             <NA>\n",
      "319       SummarizedExperiment     2021-01-27             <NA>\n",
      "320       SummarizedExperiment     2021-01-27             <NA>\n",
      "321                     DFrame     2021-01-27             <NA>\n",
      "322       SummarizedExperiment     2021-01-27             <NA>\n",
      "323           RaggedExperiment     2021-01-27             <NA>\n",
      "324           RaggedExperiment     2021-01-27             <NA>\n",
      "325           RaggedExperiment     2021-01-27             <NA>\n",
      "326       SummarizedExperiment     2021-01-27             <NA>\n",
      "327 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "328       SummarizedExperiment     2021-01-27             <NA>\n",
      "329       SummarizedExperiment     2021-01-27             <NA>\n",
      "330           RaggedExperiment     2021-01-27             <NA>\n",
      "331       SummarizedExperiment     2021-01-27             <NA>\n",
      "332       SummarizedExperiment     2021-01-27             <NA>\n",
      "333       SummarizedExperiment     2021-01-27             <NA>\n",
      "334                     DFrame     2021-01-27             <NA>\n",
      "335       SummarizedExperiment     2021-01-27             <NA>\n",
      "336           RaggedExperiment     2021-01-27             <NA>\n",
      "337           RaggedExperiment     2021-01-27             <NA>\n",
      "338           RaggedExperiment     2021-01-27             <NA>\n",
      "339       SummarizedExperiment     2021-01-27             <NA>\n",
      "340 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "341       SummarizedExperiment     2021-01-27             <NA>\n",
      "342       SummarizedExperiment     2021-01-27             <NA>\n",
      "343       SummarizedExperiment     2021-01-27             <NA>\n",
      "344           RaggedExperiment     2021-01-27             <NA>\n",
      "345       SummarizedExperiment     2021-01-27             <NA>\n",
      "346       SummarizedExperiment     2021-01-27             <NA>\n",
      "347       SummarizedExperiment     2021-01-27             <NA>\n",
      "348       SummarizedExperiment     2021-01-27             <NA>\n",
      "349       SummarizedExperiment     2021-01-27             <NA>\n",
      "350                     DFrame     2021-01-27             <NA>\n",
      "351       SummarizedExperiment     2021-01-27             <NA>\n",
      "352       SummarizedExperiment     2021-01-27             <NA>\n",
      "353       SummarizedExperiment     2021-01-27             <NA>\n",
      "354           RaggedExperiment     2021-01-27             <NA>\n",
      "355           RaggedExperiment     2021-01-27             <NA>\n",
      "356       SummarizedExperiment     2021-01-27             <NA>\n",
      "357 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "358       SummarizedExperiment     2021-01-27             <NA>\n",
      "359       SummarizedExperiment     2021-01-27             <NA>\n",
      "360           RaggedExperiment     2021-01-27             <NA>\n",
      "361       SummarizedExperiment     2021-01-27             <NA>\n",
      "362       SummarizedExperiment     2021-01-27             <NA>\n",
      "363       SummarizedExperiment     2021-01-27             <NA>\n",
      "364                     DFrame     2021-01-27             <NA>\n",
      "365       SummarizedExperiment     2021-01-27             <NA>\n",
      "366           RaggedExperiment     2021-01-27             <NA>\n",
      "367           RaggedExperiment     2021-01-27             <NA>\n",
      "368           RaggedExperiment     2021-01-27             <NA>\n",
      "369       SummarizedExperiment     2021-01-27             <NA>\n",
      "370 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "371       SummarizedExperiment     2021-01-27             <NA>\n",
      "372       SummarizedExperiment     2021-01-27             <NA>\n",
      "373           RaggedExperiment     2021-01-27             <NA>\n",
      "374       SummarizedExperiment     2021-01-27             <NA>\n",
      "375       SummarizedExperiment     2021-01-27             <NA>\n",
      "376       SummarizedExperiment     2021-01-27             <NA>\n",
      "377                     DFrame     2021-01-27             <NA>\n",
      "378       SummarizedExperiment     2021-01-27             <NA>\n",
      "379           RaggedExperiment     2021-01-27             <NA>\n",
      "380           RaggedExperiment     2021-01-27             <NA>\n",
      "381           RaggedExperiment     2021-01-27             <NA>\n",
      "382       SummarizedExperiment     2021-01-27             <NA>\n",
      "383 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "384       SummarizedExperiment     2021-01-27             <NA>\n",
      "385       SummarizedExperiment     2021-01-27             <NA>\n",
      "386           RaggedExperiment     2021-01-27             <NA>\n",
      "387       SummarizedExperiment     2021-01-27             <NA>\n",
      "388       SummarizedExperiment     2021-01-27             <NA>\n",
      "389       SummarizedExperiment     2021-01-27             <NA>\n",
      "390       SummarizedExperiment     2021-01-27             <NA>\n",
      "391                     DFrame     2021-01-27             <NA>\n",
      "392       SummarizedExperiment     2021-01-27             <NA>\n",
      "393       SummarizedExperiment     2021-01-27             <NA>\n",
      "394       SummarizedExperiment     2021-01-27             <NA>\n",
      "395           RaggedExperiment     2021-01-27             <NA>\n",
      "396           RaggedExperiment     2021-01-27             <NA>\n",
      "397       SummarizedExperiment     2021-01-27             <NA>\n",
      "398 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "399       SummarizedExperiment     2021-01-27             <NA>\n",
      "400       SummarizedExperiment     2021-01-27             <NA>\n",
      "401           RaggedExperiment     2021-01-27             <NA>\n",
      "402       SummarizedExperiment     2021-01-27             <NA>\n",
      "403       SummarizedExperiment     2021-01-27             <NA>\n",
      "404       SummarizedExperiment     2021-01-27             <NA>\n",
      "405                     DFrame     2021-01-27             <NA>\n",
      "406       SummarizedExperiment     2021-01-27             <NA>\n",
      "407           RaggedExperiment     2021-01-27             <NA>\n",
      "408           RaggedExperiment     2021-01-27             <NA>\n",
      "409           RaggedExperiment     2021-01-27             <NA>\n",
      "410       SummarizedExperiment     2021-01-27             <NA>\n",
      "411 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "412       SummarizedExperiment     2021-01-27             <NA>\n",
      "413       SummarizedExperiment     2021-01-27             <NA>\n",
      "414           RaggedExperiment     2021-01-27             <NA>\n",
      "415       SummarizedExperiment     2021-01-27             <NA>\n",
      "416       SummarizedExperiment     2021-01-27             <NA>\n",
      "417       SummarizedExperiment     2021-01-27             <NA>\n",
      "418       SummarizedExperiment     2021-01-27             <NA>\n",
      "419                     DFrame     2021-01-27             <NA>\n",
      "420       SummarizedExperiment     2021-01-27             <NA>\n",
      "421           RaggedExperiment     2021-01-27             <NA>\n",
      "422           RaggedExperiment     2021-01-27             <NA>\n",
      "423       SummarizedExperiment     2021-01-27             <NA>\n",
      "424 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "425       SummarizedExperiment     2021-01-27             <NA>\n",
      "426       SummarizedExperiment     2021-01-27             <NA>\n",
      "427           RaggedExperiment     2021-01-27             <NA>\n",
      "428       SummarizedExperiment     2021-01-27             <NA>\n",
      "429       SummarizedExperiment     2021-01-27             <NA>\n",
      "430       SummarizedExperiment     2021-01-27             <NA>\n",
      "431                     DFrame     2021-01-27             <NA>\n",
      "432       SummarizedExperiment     2021-01-27             <NA>\n",
      "433           RaggedExperiment     2021-01-27             <NA>\n",
      "434           RaggedExperiment     2021-01-27             <NA>\n",
      "435           RaggedExperiment     2021-01-27             <NA>\n",
      "436       SummarizedExperiment     2021-01-27             <NA>\n",
      "437 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "438       SummarizedExperiment     2021-01-27             <NA>\n",
      "439       SummarizedExperiment     2021-01-27             <NA>\n",
      "440       SummarizedExperiment     2021-01-27             <NA>\n",
      "441           RaggedExperiment     2021-01-27             <NA>\n",
      "442       SummarizedExperiment     2021-01-27             <NA>\n",
      "443       SummarizedExperiment     2021-01-27             <NA>\n",
      "444       SummarizedExperiment     2021-01-27             <NA>\n",
      "445       SummarizedExperiment     2021-01-27             <NA>\n",
      "446       SummarizedExperiment     2021-01-27             <NA>\n",
      "447                     DFrame     2021-01-27             <NA>\n",
      "448       SummarizedExperiment     2021-01-27             <NA>\n",
      "449       SummarizedExperiment     2021-01-27             <NA>\n",
      "450       SummarizedExperiment     2021-01-27             <NA>\n",
      "451           RaggedExperiment     2021-01-27             <NA>\n",
      "452           RaggedExperiment     2021-01-27             <NA>\n",
      "453       SummarizedExperiment     2021-01-27             <NA>\n",
      "454 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "455       SummarizedExperiment     2021-01-27             <NA>\n",
      "456       SummarizedExperiment     2021-01-27             <NA>\n",
      "457           RaggedExperiment     2021-01-27             <NA>\n",
      "458       SummarizedExperiment     2021-01-27             <NA>\n",
      "459       SummarizedExperiment     2021-01-27             <NA>\n",
      "460       SummarizedExperiment     2021-01-27             <NA>\n",
      "461                     DFrame     2021-01-27             <NA>\n",
      "462       SummarizedExperiment     2021-01-27             <NA>\n",
      "463           RaggedExperiment     2021-01-27             <NA>\n",
      "464           RaggedExperiment     2021-01-27             <NA>\n",
      "465           RaggedExperiment     2021-01-27             <NA>\n",
      "466       SummarizedExperiment     2021-01-27             <NA>\n",
      "467 RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "468       SummarizedExperiment     2021-01-27             <NA>\n",
      "469       SummarizedExperiment     2021-01-27             <NA>\n",
      "470           RaggedExperiment     2021-01-27             <NA>\n",
      "471       SummarizedExperiment     2021-01-27             <NA>\n",
      "472       SummarizedExperiment     2021-01-27             <NA>\n",
      "473       SummarizedExperiment     2021-01-27             <NA>\n",
      "474                     DFrame     2021-01-27             <NA>\n",
      "475       SummarizedExperiment     2021-01-27             <NA>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(robjects.r('''\n",
    "    curatedTCGAData(\n",
    "    diseaseCode = \"*\", assays = \"*\", version = \"2.0.1\"\n",
    ")\n",
    "'''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "b2167b59",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: snapshotDate(): 2022-04-26\n",
      "\n",
      "R[write to console]: See '?curatedTCGAData' for 'diseaseCode' and 'assays' inputs\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    ah_id                                      title file_size\n",
      "1  EH4769                       BRCA_CNASeq-20160128      0 Mb\n",
      "2  EH4770                       BRCA_CNASNP-20160128    9.8 Mb\n",
      "3  EH4771                       BRCA_CNVSNP-20160128    2.8 Mb\n",
      "4  EH4773             BRCA_GISTIC_AllByGene-20160128    1.2 Mb\n",
      "5  EH4774                 BRCA_GISTIC_Peaks-20160128      0 Mb\n",
      "6  EH4775     BRCA_GISTIC_ThresholdedByGene-20160128    0.3 Mb\n",
      "7  EH4777  BRCA_Methylation_methyl27-20160128_assays   60.7 Mb\n",
      "8  EH4778      BRCA_Methylation_methyl27-20160128_se    0.4 Mb\n",
      "9  EH4779 BRCA_Methylation_methyl450-20160128_assays 2646.4 Mb\n",
      "10 EH4780     BRCA_Methylation_methyl450-20160128_se      6 Mb\n",
      "11 EH4781                 BRCA_miRNASeqGene-20160128    0.6 Mb\n",
      "12 EH4782                    BRCA_mRNAArray-20160128   27.3 Mb\n",
      "13 EH4783                     BRCA_Mutation-20160128    4.5 Mb\n",
      "14 EH4784                  BRCA_RNASeq2Gene-20160128   43.1 Mb\n",
      "15 EH4785              BRCA_RNASeq2GeneNorm-20160128   64.5 Mb\n",
      "16 EH4786                   BRCA_RNASeqGene-20160128     30 Mb\n",
      "17 EH4787                    BRCA_RPPAArray-20160128    1.5 Mb\n",
      "                   rdataclass rdatadateadded rdatadateremoved\n",
      "1            RaggedExperiment     2021-01-27             <NA>\n",
      "2            RaggedExperiment     2021-01-27             <NA>\n",
      "3            RaggedExperiment     2021-01-27             <NA>\n",
      "4        SummarizedExperiment     2021-01-27             <NA>\n",
      "5  RangedSummarizedExperiment     2021-01-27             <NA>\n",
      "6        SummarizedExperiment     2021-01-27             <NA>\n",
      "7        SummarizedExperiment     2021-01-27             <NA>\n",
      "8        SummarizedExperiment     2021-01-27             <NA>\n",
      "9            RaggedExperiment     2021-01-27             <NA>\n",
      "10       SummarizedExperiment     2021-01-27             <NA>\n",
      "11       SummarizedExperiment     2021-01-27             <NA>\n",
      "12       SummarizedExperiment     2021-01-27             <NA>\n",
      "13       SummarizedExperiment     2021-01-27             <NA>\n",
      "14                     DFrame     2021-01-27             <NA>\n",
      "15       SummarizedExperiment     2021-01-27             <NA>\n",
      "16       SummarizedExperiment     2021-01-27             <NA>\n",
      "17       SummarizedExperiment     2021-01-27             <NA>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(robjects.r('''\n",
    "    curatedTCGAData(\n",
    "    diseaseCode = \"BRCA\", assays = \"*\", version = \"2.0.1\"\n",
    ")\n",
    "'''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "ba716867",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: snapshotDate(): 2022-04-26\n",
      "\n",
      "R[write to console]: Working on: BRCA_miRNASeqGene-20160128\n",
      "\n",
      "R[write to console]: see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation\n",
      "\n",
      "R[write to console]: downloading 1 resources\n",
      "\n",
      "R[write to console]: retrieving 1 resource\n",
      "\n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |==                                                                    |   2%\n",
      "  |                                                                            \n",
      "  |===                                                                   |   5%\n",
      "  |                                                                            \n",
      "  |=====                                                                 |   7%\n",
      "  |                                                                            \n",
      "  |=======                                                               |  10%\n",
      "  |                                                                            \n",
      "  |=========                                                             |  12%\n",
      "  |                                                                            \n",
      "  |==========                                                            |  15%\n",
      "  |                                                                            \n",
      "  |============                                                          |  17%\n",
      "  |                                                                            \n",
      "  |==============                                                        |  20%\n",
      "  |                                                                            \n",
      "  |================                                                      |  22%\n",
      "  |                                                                            \n",
      "  |=================                                                     |  25%\n",
      "  |                                                                            \n",
      "  |===================                                                   |  27%\n",
      "  |                                                                            \n",
      "  |=====================                                                 |  30%\n",
      "  |                                                                            \n",
      "  |=======================                                               |  32%\n",
      "  |                                                                            \n",
      "  |========================                                              |  35%\n",
      "  |                                                                            \n",
      "  |==========================                                            |  37%\n",
      "  |                                                                            \n",
      "  |============================                                          |  40%\n",
      "  |                                                                            \n",
      "  |==============================                                        |  42%\n",
      "  |                                                                            \n",
      "  |===============================                                       |  45%\n",
      "  |                                                                            \n",
      "  |=================================                                     |  47%\n",
      "  |                                                                            \n",
      "  |===================================                                   |  50%\n",
      "  |                                                                            \n",
      "  |=====================================                                 |  52%\n",
      "  |                                                                            \n",
      "  |======================================                                |  55%\n",
      "  |                                                                            \n",
      "  |========================================                              |  57%\n",
      "  |                                                                            \n",
      "  |==========================================                            |  60%\n",
      "  |                                                                            \n",
      "  |============================================                          |  62%\n",
      "  |                                                                            \n",
      "  |=============================================                         |  65%\n",
      "  |                                                                            \n",
      "  |===============================================                       |  67%\n",
      "  |                                                                            \n",
      "  |=================================================                     |  70%\n",
      "  |                                                                            \n",
      "  |===================================================                   |  72%\n",
      "  |                                                                            \n",
      "  |====================================================                  |  75%\n",
      "  |                                                                            \n",
      "  |======================================================                |  77%\n",
      "  |                                                                            \n",
      "  |========================================================              |  80%\n",
      "  |                                                                            \n",
      "  |==========================================================            |  82%\n",
      "  |                                                                            \n",
      "  |===========================================================           |  85%\n",
      "  |                                                                            \n",
      "  |=============================================================         |  87%\n",
      "  |                                                                            \n",
      "  |===============================================================       |  90%\n",
      "  |                                                                            \n",
      "  |=================================================================     |  92%\n",
      "  |                                                                            \n",
      "  |==================================================================    |  95%\n",
      "  |                                                                            \n",
      "  |====================================================================  |  97%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: loading from cache\n",
      "\n",
      "R[write to console]: Working on: BRCA_colData-20160128\n",
      "\n",
      "R[write to console]: see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation\n",
      "\n",
      "R[write to console]: loading from cache\n",
      "\n",
      "R[write to console]: Working on: BRCA_metadata-20160128\n",
      "\n",
      "R[write to console]: see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation\n",
      "\n",
      "R[write to console]: loading from cache\n",
      "\n",
      "R[write to console]: Working on: BRCA_sampleMap-20160128\n",
      "\n",
      "R[write to console]: see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation\n",
      "\n",
      "R[write to console]: loading from cache\n",
      "\n",
      "R[write to console]: harmonizing input:\n",
      "  removing 14736 sampleMap rows not in names(experiments)\n",
      "  removing 342 colData rownames not in sampleMap 'primary'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "df = robjects.r('''\n",
    "(BRCA_miRNASeqGene <- curatedTCGAData(\n",
    "    \"BRCA\", c(\"miRNASeqGene\"), version = \"2.0.1\", dry.run = FALSE\n",
    "))''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "bc0d1f0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A MultiAssayExperiment object of 1 listed\n",
      " experiment with a user-defined name and respective class.\n",
      " Containing an ExperimentList class object of length 1:\n",
      " [1] BRCA_CNASNP-20160128: RaggedExperiment with 1132786 rows and 2209 columns\n",
      "Functionality:\n",
      " experiments() - obtain the ExperimentList instance\n",
      " colData() - the primary/phenotype DataFrame\n",
      " sampleMap() - the sample coordination DataFrame\n",
      " `$`, `[`, `[[` - extract colData columns, subset, or experiment\n",
      " *Format() - convert into a long or wide DataFrame\n",
      " assays() - convert ExperimentList to a SimpleList of matrices\n",
      " exportClass() - save data to flat files\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(robjects.r('''\n",
    "    BRCA_CNASNP'''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "60af08e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Writing about 5 files to disk...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <span>StrVector with 5 elements.</span>\n",
       "        <table>\n",
       "        <tbody>\n",
       "          <tr>\n",
       "          \n",
       "            <td>\n",
       "            '/tmp/R...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            '/tmp/R...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            '/tmp/R...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            '/tmp/R...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            '/tmp/R...\n",
       "            </td>\n",
       "          \n",
       "          </tr>\n",
       "        </tbody>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<rpy2.robjects.vectors.StrVector object at 0x7faf9f0ac580> [RTYPES.STRSXP]\n",
       "R classes: ('character',)\n",
       "['/tmp/R..., '/tmp/R..., '/tmp/R..., '/tmp/R..., '/tmp/R...]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "robjects.r('''\n",
    "    td <- tempdir()\n",
    "tempd <- file.path(td, \"BRCA_miRNASeqGene\")\n",
    "if (!dir.exists(tempd))\n",
    "    dir.create(tempd)\n",
    "\n",
    "exportClass(BRCA_miRNASeqGene, dir = tempd, fmt = \"csv\", ext = \".csv\")\n",
    "''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "00812257",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<rpy2.robjects.methods.RS4 object at 0x7fafa579fc00> [RTYPES.S4SXP]\n",
       "R classes: ('MultiAssayExperiment',)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pandasdf = ro.conversion.rpy2py(data)\n",
    "pandasdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38e7a804",
   "metadata": {},
   "outputs": [],
   "source": [
    "from rpy2.robjects.packages import importr, data\n",
    "\n",
    "datasets = importr('datasets')\n",
    "mtcars = data(datasets).fetch('mtcars')['mtcars']\n",
    "mtcars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46ba752b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c49091b",
   "metadata": {},
   "outputs": [],
   "source": [
    "pandasdf = ro.conversion.rpy2py(mtcars)\n",
    "pandasdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c39fc44",
   "metadata": {},
   "outputs": [],
   "source": [
    "import rpy2.robjects as ro\n",
    "\n",
    "dt = pd.DataFrame()\n",
    "# To R DataFrame\n",
    "r_dt = ro.conversion.py2rpy(dt)\n",
    "# To pandas DataFrame\n",
    "pd_dt = ro.conversion.rpy2py(r_dt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "066f59e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import rpy2.robjects as ro\n",
    "from rpy2.robjects.packages import importr\n",
    "from rpy2.robjects import pandas2ri\n",
    "\n",
    "from rpy2.robjects.conversion import localconverter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f5035cb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "r_df = ro.DataFrame({'int_values': ro.IntVector([1,2,3]),\n",
    "                     'str_values': ro.StrVector(['abc', 'def', 'ghi'])})\n",
    "\n",
    "r_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "947cad40",
   "metadata": {},
   "outputs": [],
   "source": [
    "with localconverter(ro.default_converter + pandas2ri.converter):\n",
    "    pd_from_r_df = ro.conversion.rpy2py(r_df)\n",
    "\n",
    "pd_from_r_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8adef5ca",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
